File: 每日一题/数组中数字出现的次数.py
class Solution:
    def singleNumbers(self, nums: List[int]) -> List[int]:
        ret = 0
        for i in nums:
            ret ^= i
        div = 1
        while div & ret == 0:
            div <<= 1
        a = 0
        b = 0

        for i in nums:
            if div & i == 0:
                a ^= i
            else:
                b ^= i
        return [a, b]

File: 每日一题/数组中数字出现的次数.cpp
class Solution {
public:
    vector<int> singleNumbers(vector<int>& nums) {
        int ret = 0;
        for (int i : nums) {
            ret ^= i;
        }
        int div = 1;
        while ((div & ret) == 0) {
            div <<= 1;
        }
        int a = 0, b = 0;
        for (int i : nums) {
            if (div & i) {
                a ^= i;
            }
            else {
                b ^= i;
            }
        }
        return vector<int>{a, b};
    }
};